### PR TITLE
feat: add support for merging multiple configurations

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

feat/fix

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
Yes

**Summary**
- Merges all configurations together instead of single file
- Merge flag is a boolean now
- You can now do `webpack -c 1.js -c 2.js -c 3.js --merge` which was not possible before 

**Does this PR introduce a breaking change?**
No

**Other information**
/cc @evilebottnawi

### CODE REVIEW

File: packages/webpack-cli/lib/utils/cli-flags.js [hunk 1]
```diff
@@ -108,11 +108,11 @@ const core = [
...
     {
         name: 'merge',
-        usage: '--merge <path to configuration to be merged>',
+        usage: '--merge',
         alias: 'm',
-        type: String,
+        type: Boolean,
         group: CONFIG_GROUP,
-        description: 'Merge a configuration file using webpack-merge e.g. ./webpack.config.js',
+        description: 'Merge one or more configurations using webpack-merge e.g. -c ./webpack.config.js -c ./webpack.test.config.js --merge',
```
Comment (rishabh3112) [inline packages/webpack-cli/lib/utils/cli-flags.js:115]:
```suggestion
        description: 'Merge two or more configurations using webpack-merge e.g. -c ./webpack.config.js -c ./webpack.test.config.js --merge',
```

File: packages/webpack-cli/lib/groups/ConfigGroup.js [hunk 2]
```diff
@@ -190,30 +190,20 @@ class ConfigGroup extends GroupHelper {
...
-            if (!newConfigPath) {
-                throw new ConfigError("The supplied merge config doesn't exist.", 'MergeError');
+            // Get the current configuration options
+            const { options: configOptions } = this.opts;
+
+            // we can only merge when there are multiple configurations
+            // either by passing multiple configs by flags or passing a
+            // single config exporting an array
+            if (!Array.isArray(configOptions)) {
+                throw new ConfigError('No multiple configurations to be merged.', 'MergeError');
```
Comment (rishabh3112) [inline packages/webpack-cli/lib/groups/ConfigGroup.js:202]:
```suggestion
                throw new ConfigError('Atleast two configurations are required for merge', 'MergeError');
```

File: test/merge/config/merge-config.test.js [hunk 3]
```diff
@@ -1,27 +1,23 @@
 'use strict';
 
-const { stat } = require('fs');
+const { existsSync } = require('fs');
 const { resolve } = require('path');
 
 const { run } = require('../../utils/test-utils');
 
 describe('merge flag configuration', () => {
```
Comment (rishabh3112) [inline test/merge/config/merge-config.test.js:8]: Add a test case where only one config is provided. It should fail.

**Review by rishabh3112 — CHANGES_REQUESTED** [review]
Looks great!
Left a few suggestions.

**Review by rishabh3112 — APPROVED** [review]

**Review by evilebottnawi — APPROVED** [review]

**Review by snitin315 — APPROVED** [review]

**Pull request merged**